dotnet/src/Connectors/Connectors.Amazon/Bedrock/Extensions/BedrockKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Amazon.BedrockRuntime;
+using Connectors.Amazon.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.Amazon.Services;
+using Microsoft.SemanticKernel.TextGeneration;
+
+namespace Connectors.Amazon.Extensions;
+
+/// <summary>
+/// Extensions for adding Bedrock services to the application.
+/// </summary>
+public static class BedrockKernelBuilderExtensions
+{
+    /// <summary>
+    /// Add Amazon Bedrock Chat Completion service to the kernel builder using IAmazonBedrockRuntime object.
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for chat completion.</param>
+    /// <param name="bedrockApi">The IAmazonBedrockRuntime to run inference using the respective model.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockChatCompletionService(
+        this IKernelBuilder builder,
+        string modelId,
+        IAmazonBedrockRuntime bedrockApi)
+    {
+        builder.Services.AddSingleton<IChatCompletionService>(services =>
+        {
+            try
+            {
+                var logger = services.GetService<ILoggerFactory>();
+                return new BedrockChatCompletionService(modelId, bedrockApi, logger);
+            }
+            catch (Exception ex)
+            {
+                throw new KernelException($"An error occurred while initializing the BedrockChatCompletionService: {ex.Message}", ex);
+            }
+        });
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Add Amazon Bedrock Chat Completion service to the kernel builder using new AmazonBedrockRuntimeClient().
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for chat completion.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockChatCompletionService(
+        this IKernelBuilder builder,
+        string modelId)
+    {
+        // Add IAmazonBedrockRuntime service client to the DI container
+        builder.Services.TryAddAWSService<IAmazonBedrockRuntime>();
+        builder.Services.AddSingleton<IChatCompletionService>(services =>
+        {
+            try
+            {
+                var bedrockRuntime = services.GetRequiredService<IAmazonBedrockRuntime>();
```

Comment:
Let's try to set the correct user-agent string on the service client here.

Reply:
Can you ping me the commit when you set the user agent so I can compare how you do with how it is done on other libraries we create on top of the SDK?